src/sinonSandbox.js

```diff
@@ -1,25 +1,304 @@
+      (
+        path.get("callee.object").isMemberExpression() &&
+        path.get("callee.object.property").isIdentifier({ name: sinonSandboxName })
+      )
+    ) {
+      // .spy
+      if (
+        path.get("callee.property").isIdentifier({ name: "spy" }) &&
+        path.get("arguments").length === 2 &&
+        path.get("arguments.0").isIdentifier() &&
+        path.get("arguments.1").isStringLiteral()
+      ) {
+        path.get("callee").replaceWith(
+          t.memberExpression(t.identifier('jest'), t.identifier('spyOn'))
+        )
+      }
+      // .restore
+      else if (path.get("callee.property").isIdentifier({ name: "restore" })) {
+        path.getStatementParent().remove();
+      }
+    }
+  }
+});
+
+/**
+ * Handles converting sinon timers to jest timers
+ */
+const convertTimersVisitor = (clockName, isVariable, t) => ({
+  CallExpression(path) {
+    /**
+     * Handle timer tick
+     * - testContext.ss.clock.tick(100) => jest.advanceTimersByTime(100)
+     * - clock.tick(100) => jest.advanceTimersByTime(100)
+     */
+    if (
+      path.get("callee.property").isIdentifier({ name: "tick" }) &&
+      (
+        (
+          isVariable &&
+          path.get("callee.object").isIdentifier({ name: clockName })
+        ) ||
+        (
+          !isVariable &&
+          path.get("callee.object").isMemberExpression() &&
+          path.get("callee.object.property").isIdentifier({ name: clockName })
+        )
+      ) &&
+      path.get("arguments").length === 1 &&
+      path.get("arguments.0").isNumericLiteral()
+    ) {
+      path.getStatementParent().replaceWith(
+        t.expressionStatement(
+          t.callExpression(
+            t.memberExpression(
+              t.identifier("jest"),
+              t.identifier("advanceTimersByTime")
+            ),
+            path.node.arguments
+          )
+        )
+      );
+    }
+  }
+});
+
+/**
+ * Removes references to sinon
+ */
+export const removeSinonVisitor = (sinonName, t) => ({
+  CallExpression(path) {
+    /**
+     * Handle methods directly on the sinon object
+     * - sinon.createSandbox()
+     * - sinon.createSandbox({ useFakeTimers: true })
+     * - sinon.createSandbox({ useFakeTimers: true, useFakeServer: true })
+     * - sinon.useFakeTimers()
+     */
+    // sinon.
+    if (path.get("callee").isMemberExpression() && path.get("callee.object").isIdentifier({ name: sinonName })) {
+      // .createSandbox()
+      if (path.get("callee.property").isIdentifier({ name: "createSandbox" })) {
+        // Remove references to the sinon sandbox timer if useFakeTimers: true
+        if (
+          path.get("arguments").length > 0 &&
+          path.get("arguments.0").isObjectExpression() &&
+          path.node.arguments[0].properties.findIndex((prop) => prop.key.name === "useFakeTimers") !== -1
+        ) {
+          path.findParent((parentPath) => parentPath.isProgram()).traverse(convertTimersVisitor("clock", false, t));
```

Comment:
Why we need "clock" here?

Reply:
When `useFakeTimers` is used, then the sandbox will have a property called `clock` that refers to the mock timer